client/js/lounge.js

```diff
@@ -276,10 +278,29 @@ $(function() {
 		return msg;
 	}
 
-	function buildChannelMessages(channel, messages) {
-		return messages.reduce(function(docFragment, message) {
-			docFragment.append(buildChatMessage({
-				chan: channel,
+	function appendMessage(container, chan, type, msg) {
+		if ($(msg).is(".join, .part, .quit, .mode, .kick, .nick") && type !== "lobby") {
```

Comment:
Ah wait, the message we get through is the dom element anyway. We don't get the information of the message without it being an element.